src/logic/safe/store/reducer/gatewayTransactions.ts

```diff
@@ -89,8 +118,8 @@ export const gatewayTransactionsReducer = handleActions<GatewayTransactionsState
     },
     [ADD_QUEUED_TRANSACTIONS]: (state, action: Action<QueuedPayload>) => {
       const { chainId, safeAddress, values } = action.payload
-      let newNext = {}
-      let newQueued = {}
+      let newNext = cloneDeep(state[chainId]?.[safeAddress]?.queued?.next || {})
+      let newQueued = cloneDeep(state[chainId]?.[safeAddress]?.queued?.queued || {})
```

Comment:
Same here, no need to do a deep copy.

Reply:
I found a bug whereby we weren't removing transactions that move from `queued.queued` to `queued.next`. We will have to still clone `queued.queued` in this instance.

Reply:
Shallow copy would be enough.

Reply:
True. I just tested it